packages/grid/test/keyboard-navigation.test.js

```diff
@@ -1934,4 +1938,162 @@ describe('keyboard navigation on column groups', () => {
+        await nextFrame();
+
+        const tabbableHeaderCell = getTabbableCells(header)[0];
+        expect(tabbableHeaderCell.offsetHeight).not.to.equal(0);
+        expect(tabbableHeaderCell).not.to.equal(initialTabbableHeaderCell);
+      });
+
+      it('should have no tabbable header cells when header is hidden', async () => {
+        // Hide all header rows
+        mainGroup.headerRenderer = null;
+        subGroup.headerRenderer = null;
+        column.headerRenderer = null;
+        await nextFrame();
+
+        const tabbableHeaderCell = getTabbableCells(header)[0];
+        expect(tabbableHeaderCell).not.to.be.ok;
+      });
+
+      it('should update tabbable header cell on header row unhide', async () => {
+        // Hide all header rows
+        mainGroup.headerRenderer = null;
+        subGroup.headerRenderer = null;
+        column.headerRenderer = null;
+        await nextFrame();
+
+        column.header = 'column';
+        await nextFrame();
+
+        const tabbableHeaderCell = getTabbableCells(header)[0];
+        expect(tabbableHeaderCell.offsetHeight).not.to.equal(0);
+      });
+    });
+
+    describe('updating tabbable cells - body', () => {
+      let body;
+
+      beforeEach(() => {
+        body = grid.$.items;
+      });
+
+      it('should update tabbable body cell on body row hide', async () => {
+        // Focus the second body row / make it tabbable
+        tabToBody();
+        down();
+
+        // Hide the second body row
+        grid.items = ['foo'];
+
+        await nextFrame();
+
+        // Expect the tabbable body cell to be on the first row
+        const tabbableBodyCell = getTabbableCells(body)[0];
+        expect(tabbableBodyCell.parentElement.index).to.equal(0);
+        expect(tabbableBodyCell.offsetHeight).not.to.equal(0);
+      });
+
+      it('should have no tabbable body cell when there are no rows', async () => {
+        // Remove all body rows
+        grid.items = [];
+
+        const tabbableBodyCell = getTabbableCells(body)[0];
+        expect(tabbableBodyCell).not.to.be.ok;
+      });
+
+      it('should update tabbable body cell on body row unhide', async () => {
+        // Remove all body rows
+        grid.items = [];
+        await nextFrame();
+
+        grid.items = ['foo', 'bar'];
+        await nextFrame();
+
+        const tabbableBodyCell = getTabbableCells(body)[0];
+        expect(tabbableBodyCell.parentElement.index).to.equal(0);
+        expect(tabbableBodyCell.offsetHeight).not.to.equal(0);
+      });
+
+      it('should remain on the first column after a tabbable cell row is hidden', async () => {
+        // Add a second column
+        const column = document.createElement('vaadin-grid-column');
+        grid.appendChild(column);
+        await nextFrame();
+
+        // Focus the second cell on the second row
+        tabToBody();
+        right();
+        down();
+
+        // Hide the second body row
+        grid.items = ['foo'];
```

Comment:
It made me have to scroll to the top in order to check what were the initial items and make sure there is more than one of them. Maybe using something around `pop()` could make the underlying idea more obvious:

```suggestion
        grid.items.pop();
        grid.items = grid.items;
```

Anyway, I put it here more for the note, it is not that I'm convinced that `pop()` would make it better.

Reply:
Used `grid.items = [grid.items[0]];`, doesn't have to be same array instance